### PR TITLE
Add menu bar button for media embed.

### PR DESCRIPTION
**Pull request by mremiszewski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Feature (media-embed): Added menu bar integration to media embed.

MINOR BREAKING CHANGE (media-embed): Media embed uses dialog instead of dropdown for inserting media.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-media-embed/src/mediaembedui.ts [hunk 1]
```diff
@@ -40,74 +39,95 @@ export default class MediaEmbedUI extends Plugin {
...
 
-		editor.ui.componentFactory.add( 'mediaEmbed', locale => {
-			const dropdown = createDropdown( locale );
+		editor.ui.componentFactory.add( 'mediaEmbed', () => {
+			const button = this._createDialogButton( ButtonView );
 
-			this._setUpDropdown( dropdown, command );
+			button.set( {
+				tooltip: true
+			} );
```
Comment (DawidKossowski) [inline packages/ckeditor5-media-embed/src/mediaembedui.ts:48]:
```suggestion
			button.tooltip = true;
```

When we set only one value we rather use the shorthand form.

File: packages/ckeditor5-media-embed/src/mediaembedui.ts [hunk 2]
```diff
@@ -40,74 +39,95 @@ export default class MediaEmbedUI extends Plugin {
...
-			form.urlInputView.bind( 'isEnabled' ).to( command, 'isEnabled' );
-		} );
+	private _showDialog() {
+		const editor = this.editor;
+		const dialog = editor.plugins.get( 'Dialog' );
+		const command = editor.commands.get( 'mediaEmbed' )!;
+		const t = editor.locale.t;
 
-		dropdown.bind( 'isEnabled' ).to( command );
+		// if ( !this.formView ) {
```
Comment (DawidKossowski) [inline packages/ckeditor5-media-embed/src/mediaembedui.ts:96]: Why did you comment it out? I believe we do not want to create a new view each time.

**Review by DawidKossowski — CHANGES_REQUESTED** [review]
There are still several things to fix/improve:

1. You should add the `Dialog` plugin to `requires` in `mediaembedui.ts`.
2. IMO, the dialog is too small. We could discuss it at the daily meeting.
3. The dialog content does not clear after closing, resulting in the bug where the input label hovers over the starting input value (after reopening the modal):

https://github.com/ckeditor/ckeditor5/assets/20561161/0327fea4-e5c5-4ac2-a20a-0ea8db34e033

4. Unit tests are still missing.

**Comment by mremiszewski** [comment]
Ad 3. is still not fixed, I'm waiting for a decision about the way it will be handled.

File: packages/ckeditor5-media-embed/tests/mediaembedui.js [hunk 3]
```diff
@@ -35,21 +32,12 @@ describe( 'MediaEmbedUI', () => {
...
-				dropdown.isOpen = false;
-
-				form = dropdown.panelView.children.get( 0 );
+				button = editor.ui.componentFactory.create( 'mediaEmbed' );
 			} );
 	} );
 
 	afterEach( () => {
-		dropdown.element.remove();
+		// button.element.remove();
```
Comment (scofalik) [inline packages/ckeditor5-media-embed/tests/mediaembedui.js:40]: Remove this line if it is not needed.

File: packages/ckeditor5-media-embed/tests/mediaembedui.js [hunk 4]
```diff
@@ -59,218 +47,182 @@ describe( 'MediaEmbedUI', () => {
...
 
-		it( 'should add a form to the panelView#children collection', () => {
-			expect( dropdown.panelView.children.length ).to.equal( 1 );
-			expect( dropdown.panelView.children.get( 0 ) ).to.be.instanceOf( MediaFormView );
+	describe( 'menuBar button', () => {
+		beforeEach( () => {
+			button = editor.ui.componentFactory.create( 'menuBar:mediaEmbed' );
 		} );
+		afterEach( () => {
+			// menuBarButton.element.remove();
```
Comment (scofalik) [inline packages/ckeditor5-media-embed/tests/mediaembedui.js:72]: Same here, remove if not needed.

**Review by scofalik — CHANGES_REQUESTED** [review]
In general LGTM. I tested it manually a bit and it looks good. Mostly minor changes except that it should be a centered modal.

**Review by scofalik — CHANGES_REQUESTED** [review]
One minor thing. I'd change it myself but there may be a related unit test. If there's not maybe we should check how the modal is opened (with what params?)

File: packages/ckeditor5-media-embed/src/mediaembedui.ts [hunk 5]
```diff
@@ -95,14 +93,16 @@ export default class MediaEmbedUI extends Plugin {
...
+
 			this._formView = new ( CssTransitionDisablerMixin( MediaFormView ) )( getFormValidators( editor.t, registry ), editor.locale );
 		}
 
 		dialog.show( {
 			id: 'mediaEmbed',
 			title: t( 'Insert media' ),
 			content: this._formView,
-			position: DialogViewPosition.EDITOR_TOP_SIDE,
+			position: DialogViewPosition.EDITOR_CENTER,
```
[the remark this answers is not in the document]
Comment (mremiszewski) [reply packages/ckeditor5-media-embed/src/mediaembedui.ts]: Done

**Pull request merged**